### PR TITLE
feat: add conversation caching

### PR DESCRIPTION
**Pull request by rygine**
## What does this PR do?

Add conversation caching for DMs to drastically speed up conversation loading, especially for users with tons of conversations.

Close ENG-186

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Load your messages page and confirm that loading of conversations is faster.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/src/store/message-cache.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import { persist } from 'zustand/middleware';
+
+// If any breaking changes to the ConversationExport schema occur, increment the cache version.
+const CONVERSATION_CACHE_VERSION = 1;
+
+/**
+ * The MessageCache is a JSON serializable Zustand store that is persisted to LocalStorage
+ * Persisting conversations to the cache saves on both bandwidth and CPU cycles, as we don't have to re-fetch or re-decrypt conversations on subsequent page loads
+ */
+interface MessageCache {
```
Comment (neekolas) [inline apps/web/src/store/message-cache.ts:12]: The name `MessageCache` feels a little misleading, seeing as it doesn't store any messages

Comment (rygine) [reply apps/web/src/store/message-cache.ts]: makes sense, addressed here: https://github.com/lensterxyz/lenster/pull/1908/commits/2b5d5f8f34642b7198e651786638d744c9fbdb7e

**Comment by bigint** [comment]
https://user-images.githubusercontent.com/69431456/223197789-5c205215-b69a-47b2-a86e-fb22e7a99bc9.mov

Still too much time?? no?

**Comment by rygine** [comment]
> Still too much time?? no?

just to confirm, this video is after the cache is warm?

if so, what's the time difference between the current production build and this cached version?

**Comment by bigint** [comment]
@rygine It reduced to half

30s on main branch
15s in this branch

**Comment by bigint** [comment]
It's great improvement 👏🏼 

But looking forward for improvement to the core APIs 🙇🏼 

Gonna merge this now 🚀

**Review by bigint — APPROVED** [review]
**LGTM** :shipit: 

Rad LGTM

**Pull request merged**